charts/microgateway/README.md.gotmpl

```diff
@@ -1,153 +1,365 @@
-More details about the installation of the microgateway helm chart can be found in chapter [Helm basics](#helm-basics).
+## Quick start guide
 
-## Adding the Repository
+The following subchapters describe how to use the Helm chart.
+
+### Adding the chart repository
 
 To add the chart repository:
 
 ```console
 helm repo add airlock https://ergon.github.io/airlock-helm-charts/
 ```
 
-## Installing the Chart
+### Installing the chart
+
+To install the chart with the release name `microgateway`:
+
+```console
+helm upgrade -i microgateway airlock/microgateway
+```
+
+### Uninstalling the chart
+
+To uninstall the chart with the release name `microgateway`:
+
+```console
+helm uninstall microgateway
+```
+
+## Configuration
+
+The following table lists configuration parameters of the Airlock Microgateway chart and the default values.
+
+{{ template "chart.valuesTable" . }}
+
+## Getting started with Helm
+
+This chapter provides information to get started with Helm.
+
+### Override the default values
+
+The Airlock Microgateway Helm chart has many parameters and most of them have default values (see [Configuration](#configuration)). Depending on the environment, the defaults must be adapted. To override the default values, do the following:
+
+* Create a yaml file which contains the values differ to the default.
+* Apply this yaml file with the `-f` parameter.
+
+**Example:**
+
+The example below shows how certain default values could be adjusted.
+
+  custom-values.yaml
+  ```
+  config:
+    default:
+      backend:
+        hostname: other-service
+    existingSecret: "microgateway-secrets"
+  imagePullSecrets:
+    - name: "docker-secret"
+  ingress:
+    enabled: true
+    hosts:
+      - example.com
+  ```
+
+Afterwards apply the Helm chart configuration file with the `-f` parameter.
+```console
+helm upgrade -i microgateway airlock/microgateway -f custom-values.yaml
+```
+
+:point_up: **YAML indentation**:<br>
+YAML is very strict with indetation. To ensure that the YAML file itself is correct, check it's content with a YAML validator (e.g. [YAML Lint](http://www.yamllint.com/)).
 
-To install the chart with the release name `waf`:
+## Dependencies
+
+The Airlock Microgateway Helm chart has the following optional dependencies, which can be enabled for an easy start.
+
+{{ template "chart.requirementsTable" . }}
+
+### Redis
+
+In case that sessionhandling is enabled on Airlock Microgateway, a Redis service needs to be available. The Airlock Microgateway Helm chart has two options:
+* To deploy the dependend Redis service, adapt the Helm chart configuration as shown below:
+  custom-values.yaml
+  ```
+  redis:
+    enabled: true
+  ```
+* To use an existing Redis service, adapt the Helm chart configuration as shown below:
+  custom-values.yaml
+  ```
+  config:
+    redisService:
+      - <REDIS-SERVICE1>:<PORT>
+      - <REDIS-SERVICE2>:<PORT>
+  redis:
+    enabled: false
```

Comment:
Unnecessary configuration. 
I would mention this in the documentary and not in the values.yaml.
e.g 
If an own redis is used, the setting redis.enabled is not necessary.

Reply:
I know that it is unnecessary. I think it's much better to illustrate in a example than in a long sentence nobody reads. What do you think?

Reply:
Okay.